jsonschema-core.md

```diff
@@ -1045,7 +1045,10 @@ IRI.
 
 Note that this IRI is an identifier and not necessarily a network locator. In
 the case of a network-addressable URL, a schema need not be downloadable from
-its canonical IRI.
+its canonical IRI. However, schema authors SHOULD refrain from creating schemas
+that use a locatable `$id` value unless the schema will be available at the
+indicated location. Otherwise, an IRI which does not convey location semantics
+SHOULD be used.
```

Comment:
I'm still iffy on including this.  We don't have much in the way for recommendations for schema authors mainly because we really can't control what they do.

Reply:
I'm not convinced we should include this in the spec either.
I think this is the only section which strays into best practice though.

> I also don't think this recommendation makes sense given the direction these changes and the discussions behind them have gone.

I think we have (speaking personally) intrenched to almost always use https, that using a non-locating semantics URI scheme seems strange. But I do think it's something we should strive to wards suggesting as best practice. It would avoid confusion, as you've said. But I also think we shouldn't allow things to be required that way. (Won't say any more on the topic here.)

I think we can/should track potential best practice recomendations as Issues in the Community repo, and maybe migrating them to the new website repo when appropriate.